src/iqm/cirq_iqm/serialize.py

```diff
@@ -30,8 +32,25 @@ def serialize_circuit(circuit: Circuit) -> iqm_client.Circuit:
     Returns:
         data transfer object representing the circuit
     """
-    instructions = tuple(map(map_operation, circuit.all_operations()))
-    return iqm_client.Circuit(name='Serialized from Cirq', instructions=instructions)
+    total_ops_list = [op for moment in circuit for op in moment]
+    instructions = list(map(map_operation, total_ops_list))
+    for idx, op in enumerate(total_ops_list):
+        if isinstance(op, cirq.ClassicallyControlledOperation):
+            feedback_key = str(op._conditions[0].keys[0])
+            measurement_already_used = False
+            for m_idx, i in enumerate(instructions):
+                if i.name == "measure" and i.args["key"] == feedback_key:
+                    if m_idx > idx:
+                        raise OperationNotSupportedError("Measurement condition must precede cc_prx operation")
+                    if measurement_already_used:  # raise error if measurement has already been used as condition
+                        raise OperationNotSupportedError("Measurement condition for cc_prx must only be from one qubit")
+                    measurement_already_used = True  # change the flag to True now that measurement is used
+                    if len(i.qubits) > 1:
+                        raise OperationNotSupportedError("Measurement condition for cc_prx must only be from one qubit")
+                    feedback_qubit = i.qubits[0]
+                    instructions[idx].args["feedback_key"] = feedback_key
+                    instructions[idx].args["feedback_qubit"] = feedback_qubit
```

Comment:
I think you can avoid this double loop if you build a dict from measurement keys to the measurement ops, and a set of used keys.

Reply:
Then you can only iterate through the circuit once.